torch/_decomp/decompositions.py

```diff
@@ -2010,37 +2015,53 @@ def upsample_bilinear2d(
     out_w = sym_float(output_size[1])
 
     # Calculate horizontal and vertical scaling factor
-    # TODO: Figure out if scales_h/scales_w matters here
+    # See compute_scales_value and area_pixel_compute_scale in core.
     if out_h > 1:
         if align_corners:
             h_scale_factor = (in_h - 1) / (sym_int(out_h) - 1)
         else:
-            h_scale_factor = in_h / out_h
+            if scales_h is not None and scales_h > 0:
+                h_scale_factor = 1.0 / scales_h
```

Comment:
This was missing and resulted in numerical differences, see `compute_scales_value`